### PR TITLE
standardise start function

### LINKED ISSUE #5: Some API methods have an optional last parameter making them difficult to extend
Hi @pdincau 

In creating the last PR (for parameterized jobs) it would have been nicer to just extend the existing `Jobs.start` function (currently it is like this)
```elixir
def start(job, token \\ ExJenkins.token) do
    post("job/" <> job <> "/build?token=" <> token, "")
    |> handle_start_job_response
end
```
to add the Jenkins build parameters as an additional argument, however this was not possible as the last argument is optional.

Can I suggest that the signature of functions are changed to pass optional arguments as options instead?

Although this is a breaking change, it would allow the functions to be extended more easily without further breaking changes.

So the `Jobs.start` function would become:

```elixir
def start(job, opts \\ []) do
    token = Keyword.get(opts, :token, ExJenkins.token)
    post("job/" <> job <> "/build?token=" <> token, "")
    |> handle_start_job_response
end
```
This would then be invoked as
```elixir
start("my-job", token: "my-token")
```

This would allow another option to be passed `:build_parameters` without having to have a separate `build_with_parameters` function.

I would like to enhance the `Jobs.status` function to respond with additional information if an additional option is passed (the parameters used for a parameterized build), but I cannot do this without creating a new function at the moment.

Thanks

Charles

### PR DESCRIPTION
**Pull request by pdincau**
@chazsconi this should fix https://github.com/pdincau/ex_jenkins/issues/5

### CODE REVIEW

File: lib/ex_jenkins/endpoints/jobs.ex [hunk 1]
```diff
@@ -10,29 +10,30 @@ defmodule ExJenkins.Jobs do
...
+        iex> ExJenkins.Jobs.start("myjob", params: [foo: "bar", foo2: "bar2"], token: "anothertoken")
         {:ok, {:started, location}}
   """
-  def start_with_parameters(job, params, token \\ ExJenkins.token) do
-    post("job/" <> job <> "/buildWithParameters?token=" <> token, {:form, params})
-    |> handle_start_job_response
+  def start(job, opts \\ []) do
+    token = Keyword.get(opts, :token, ExJenkins.token)
+    params = Keyword.get(opts, :params, [])
+    response = case params do
```
Comment (chazsconi) [inline lib/ex_jenkins/endpoints/jobs.ex:30]: If you default the params to `nil` (or in fact provide no default as that will be `nil`) then you can just use an `if` and not a `case`.

Comment (chazsconi) [reply lib/ex_jenkins/endpoints/jobs.ex:30]: But not really a major issue - just a minor point of style

**Review by chazsconi — APPROVED** [review]

File: lib/ex_jenkins/endpoints/jobs.ex (hunk 1 above)
Comment (pdincau) [reply lib/ex_jenkins/endpoints/jobs.ex:30]: yep, but i'm not very much a fan of the "if". Thank you for pointing out though 👍

Comment (chazsconi) [reply lib/ex_jenkins/endpoints/jobs.ex:30]: No problem!  Cool that you made the change 👍

**Pull request merged**